dev/pastetools/getclipboard.html

```diff
@@ -0,0 +1,309 @@
+		} );
+
+		if ( oldInstance ) {
+			oldInstance.once( 'destroy', function() {
+				setTimeout( createEditor, 500 );
+			} );
+
+			return oldInstance.destroy( true );
+		}
+
+		createEditor();
+
+		function createEditor() {
+			var textarea = new CKEDITOR.dom.element( 'textarea' ),
+				allowedContent = CKEDITOR.document.getById( 'allowedContent' ),
+				extraAllowedContent = CKEDITOR.document.getById( 'extraAllowedContent' ),
+				config = {
+					language: 'en',
+					embed_provider: 'https://ckeditor.iframe.ly/api/oembed?url={url}&callback={callback}',
+					cloudServices_uploadUrl: 'https://33333.cke-cs.com/easyimage/upload/',
+					// Note: this is a token endpoint to be used for CKEditor 4 samples / developer tests only. Images uploaded using the testing token service may be deleted automatically at any moment.
+					// To create your own token URL please visit https://ckeditor.com/ckeditor-cloud-services/.
+					cloudServices_tokenUrl: 'https://33333.cke-cs.com/token/dev/ijrDsqFix838Gh3wGO3F77FSW94BwcLXprJ4APSp3XQ26xsUHTi0jcb1hoBt',
+					mathJaxLib: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS_HTML',
+					plugins: plugins
+				};
+
+			if ( allowedContent.$.checked ) {
+				config.allowedContent = true;
+			}
+
+			if ( extraAllowedContent.getValue() ) {
+				config.extraAllowedContent = extraAllowedContent.getValue();
+			}
+
+			editorContainer.setHtml( '' );
+			editorContainer.append( textarea );
+
+			var editor = CKEDITOR.replace( textarea, config );
+
+			editor.on( 'paste', function( evt ) {
+				// Clipboard text/html processing.
+				var val = evt.data.dataValue;
+				if ( evt.data.dataTransfer && evt.data.dataTransfer.getData( 'text/html', true ) ) {
+					val = evt.data.dataTransfer.getData( 'text/html', true );
+					document.getElementById( 'emHtml' ).innerHTML = '<br />Data from: <code>dataTransfer.getData( \'text/html\', true )</code>.';
+				} else {
+					document.getElementById( 'emHtml' ).innerHTML = '<br /><code>dataTransfer.getData( \'text/html\', true )</code> was empty.' +
+					'<br /><span class="marker">Data from: <code>evt.data.dataValue</code>.</span>';
+				}
+				document.getElementById( 'rawHtml' ).value = val;
+
+				// Clipboard text/rtf processing.
+				val = '';
+				if ( evt.data.dataTransfer && evt.data.dataTransfer.getData( 'text/rtf' ) ) {
+					val = evt.data.dataTransfer.getData( 'text/rtf', true );
+					document.getElementById( 'emRtf' ).innerHTML = '<br />Data from: <code>dataTransfer.getData( \'text/rtf\', true )</code>.';
+				} else {
+					document.getElementById( 'emRtf' ).innerHTML = '<br /><code>dataTransfer.getData( \'text/rtf\', true )</code> was empty';
+				}
+				document.getElementById( 'rawRtf' ).value = val;
+
+			}, null, null, -1 );
+
+			editor.on( 'paste', function( evt ) {
+				setTimeout( function() {
+					document.getElementById( 'output' ).value = editor.getData();
+				}, 0 );
+			}, null, null, 999 );
+		}
+	}
```

Comment:
This code is unreadable - I know it's mostly copy&paste from the previous example, but maybe you could extract some common variables (like `document.getElementById( 'emHtml' )`) and overall refactor it a bit, so we are the last one who needs to understand this complicated code?